.github/workflows/ci.yaml

```diff
@@ -603,16 +613,27 @@ jobs:
         with:
           key-prefix: test-go-race-pg-${{ runner.os }}-${{ runner.arch }}
 
+      - name: Normalize Terraform Path for Caching
+        shell: bash
+        # Terraform gets installed in a random directory, so we need to normalize
+        # the path or many cached tests will be invalidated.
+        run: |
+          mkdir -p "$RUNNER_TEMP/sym"
+          source scripts/normalize_path.sh
+          normalize_path_with_symlinks "$RUNNER_TEMP/sym" "$(dirname "$(which terraform)")"
+
       # We run race tests with reduced parallelism because they use more CPU and we were finding
       # instances where tests appear to hang for multiple seconds, resulting in flaky tests when
       # short timeouts are used.
       # c.f. discussion on https://github.com/coder/coder/pull/15106
+      # Our Linux runners have 32 cores, but we use reduced parallelism for race detection.
       - name: Run Tests
-        env:
-          POSTGRES_VERSION: "17"
-        run: |
-          make test-postgres-docker
-          gotestsum --junitfile="gotests.xml" --packages="./..." -- -race -parallel 4 -p 4
+        uses: ./.github/actions/test-go-pg
+        with:
+          postgres-version: "17"
+          test-parallelism-packages: "16"
+          test-parallelism-tests: "8"
```

Comment:
it was 4x4 with 16 cores, or 1 per core.

now it's 16x8 with 32 cores, or 4 per core. This is likely to make flakes worse.